implementation/src/main/java/io/smallrye/config/ConfigValueConfigSource.java

```diff
@@ -401,14 +405,15 @@ private void load0(LineReader lr) throws IOException {
                 }
                 String key = loadConvert(lr.lineBuf, 0, keyLen, convtBuf);
                 String value = loadConvert(lr.lineBuf, valueStart, limit - valueStart, convtBuf);
-                put(key, ConfigValue.builder()
-                        .withName(key)
-                        .withValue(value)
-                        .withRawValue(value)
-                        .withConfigSourceName(configSourceName)
-                        .withConfigSourceOrdinal(configSourceOrdinal)
-                        .withLineNumber(lr.lineNumber)
-                        .build());
+                this.checkPutResult.accept(
+                        put(key, ConfigValue.builder()
```

Comment:
`.put` returns the previous object associated with the key. It can easily be used to check and warn about the old and new values.